src/ceph-disk/ceph_disk/main.py

```diff
@@ -3634,6 +3634,10 @@ def find_cluster_by_uuid(_uuid):
         else:
             if fsid == _uuid:
                 return cluster
+            elif fsid == '00000000-0000-0000-0000-000000000000':
```

Comment:
@dmick can we just do this check in `get_fsid()`? as `get_conf_with_default()` never returns a `None`.  i think it'd be better to `s/if fsid is None/if fsid == '00000000-0000-0000-0000-000000000000'/` in `get_fsid()`.

Reply:
I don't have strong feelings about it.  This way, it's more resilient to different return values from Ceph, in case anyone ever changed/changes their minds about it, and this way, no other callers of get_fsid() need to be validated and possibly changed, so it seemed like "least change"; since ceph-disk is being deprecated, I leaned toward "least possible risk".

Reply:
> This way, it's more resilient to different return values from Ceph, in case anyone ever changed/changes their minds about it

i think it's more about how `get_fsid()` should behave. because, again, `get_conf_with_default()` never returns a `None`. and `get_fsid()` is expected to return a valid fsid. the latter is more important. as we persist the return value of it as`ceph_fsid`, which will read back and is compared with the return value of `get_fsid()` when we activate an OSD.

so it'd be better to fail `ceph-disk` sooner if we have an empty fsid.

> I leaned toward "least possible risk".

so do i. but i prefer correctness to it.

Reply:
I don't understand the argument, but I can change the code.  It's nearly dead anyway.